### ISSUE TITLE
Arch Linux compatiblity

### ISSUE THREAD

**Issue by NicoHood** [hyperion-project/hyperion#696]
As I now made the switch to arch, I also want hyperion running for sure. But the installation script fails, as apt-get for example does not even exist. And the service seems to be installed wrong.

For now I suggest to add a check which os is running. If its != raspbian you should abort.

And for the `make install` thing I suggest to also install the service there, as this is the way to install software from git. A debian package does not really work good on arch.

I will try and test more the next days.

**Comment by redPanther** [comment]
I also use arch. I never use or even think to use the install script. Those things usualy not work. Make install will not install the service.
My plan is to make a AUR script. This will do distro integration.
Arch is not for noobs. For arch users I must expect that they will do things on their own.

**Comment by patlux** [comment]
There are 4 packages for arch:

https://aur.archlinux.org/packages/?O=0&K=hyperion

For raspberry install [hyperion-rpi-bin](https://aur.archlinux.org/packages/hyperion-rpi-bin/):

**via [pacaur](https://github.com/rmarquis/pacaur)**
```bash
pacaur -S hyperion-rpi-bin
```

**via [yaourt](https://github.com/archlinuxfr/yaourt)**
```bash
yaourt -S hyperion-rpi-bin
```

**[manual](https://wiki.archlinux.org/index.php/Arch_User_Repository#Installing_packages):**
```bash
curl "https://aur.archlinux.org/cgit/aur.git/snapshot/hyperion-rpi-bin.tar.gz" |tar xz
cd "hyperion-rpi-bin"
makepkg -si
```

**Comment by BadderBlackBart** [comment]
@patlux When I try to build the 'hyperion-rpi-bin' package using the manual method, I get a 404 error on 'https://raw.githubusercontent.com/tvdzwan/hyperion/V1.01.1/deploy/hyperion_rpi.tar.gz'. 
Using Arch on RPI3

**Comment by redPanther** [comment]
The 'packages' are outdated. They have to be reworked using new 'make install', new git url, new location in filesystem and new cmake flags.

**Comment by NicoHood** [comment]
I suggest to add information to this new feature to the instructions text, as this https://github.com/hyperion-project/hyperion/pull/641#issuecomment-220803746 is now closed.

**Comment by redPanther** [comment]
files hosted on sourceforge:
development version:
https://sourceforge.net/projects/hyperion-project/files/beta

stable version:
https://sourceforge.net/projects/hyperion-project/files/release

**Comment by NicoHood** [comment]
Those precompiled versions also contain some personal information:
```
Device unblocked
Opening UART: /dev/ttyACM0
Unable to open RS232 device (IO Exception (2): No such file or directory, file /home/sascha/hyperion/dependencies/build/serial/src/impl/unix.cc, line 100.)
```

**Comment by patlux** [comment]
@BadderBlackBart: Oh sorry, I didn't see the changes at all. I will update my packages in the aur in the next few days..

In the meantime use the attached `PKGBUILD`.
[PKGBUILD.txt](https://github.com/hyperion-project/hyperion/files/313846/PKGBUILD.txt)

```bash
wget https://github.com/hyperion-project/hyperion/files/313846/PKGBUILD.txt
mv PKGBUILD.txt PKGBUILD
makepkg -si
```

I tested it on my raspberry pi 2 and its working.

```
Hyperion Ambilight Deamon (7157)
        Version   : V1.03.0 (GitHub-b8f7642/c5d5f2d-1465599599
        Build Time: Jun 11 2016 01:39:39
Hyperion Daemon
Build time: Jun 11 2016 01:39:12
```

Have fun.

**Comment by NicoHood** [comment]
I've updated the AUR package now.
https://aur.archlinux.org/packages/hyperion/

This can be closed. I am going to update the other packages over the next days hopefully.

**Comment by NicoHood** [comment]
However I get those errors with the latest master:

```
/bin/sh: --cpp_out: command not found
make[2]: *** [libsrc/protoserver/CMakeFiles/protoserver.dir/build.make:78: libsrc/protoserver/message.pb.cc] Error 127
make[2]: *** Waiting for unfinished jobs....
[ 67%] Built target xbmcvideochecker
[ 69%] Built target framebuffer-grabber
[ 70%] Generating __/__/__/include/grabber/moc_X11Grabber.cxx
[ 75%] Built target jsonserver
[ 79%] Built target v4l2-grabber
[ 83%] Built target boblightserver
/AUR/hyperion-git/src/hyperion-git/include/grabber/X11Grabber.h:0: Note: No relevant classes found. No output generated.
make[1]: *** [CMakeFiles/Makefile2:647: libsrc/protoserver/CMakeFiles/protoserver.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
Scanning dependencies of target x11-grabber
[ 84%] Building CXX object libsrc/grabber/x11/CMakeFiles/x11-grabber.dir/X11Grabber.cpp.o
[ 84%] Building CXX object libsrc/grabber/x11/CMakeFiles/x11-grabber.dir/__/__/__/include/grabber/moc_X11Grabber.cxx.o
[ 84%] Linking CXX static library ../../../lib/libx11-grabber.a
[ 84%] Built target x11-grabber
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

With this PKGBUILD:
https://gist.github.com/NicoHood/4a020d59e30833da9e23bb1e80ab5c42

**Comment by NicoHood** [comment]
I also got this error with the tagged release. Any idea why?

```
[ 55%] Built target jsonserver
[ 56%] Running C++ protocol buffer compiler on /home/arch/hackallthethings/Arch-Scripts/AUR/hyperion/src/hyperion-1.03.2/libsrc/protoserver/message.proto
/bin/sh: --cpp_out: command not found
make[2]: *** [libsrc/protoserver/CMakeFiles/protoserver.dir/build.make:78: libsrc/protoserver/message.pb.cc] Error 127
make[1]: *** [CMakeFiles/Makefile2:650: libsrc/protoserver/CMakeFiles/protoserver.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

**Comment by redPanther** [comment]
Please verify that you can compile without using a pkg biuilf. Checkout from scratch and do your build steps.

**Comment by NicoHood** [comment]
I tried the current master branch now with:
```
cmake -DCMAKE_INSTALL_PREFIX="${pkgdir}/usr" -DPLATFORM=x86 -DCMAKE_BUILD_TYPE=Release \
				-DENABLE_QT5=ON -DUSE_SHARED_AVAHI_LIBS=ON -DUSE_SYSTEM_PROTO_LIBS=ON ..
```
Still got the same error. Also with the submodules enabled. Looks like a possible missing dep? Or a different syntax for a newer arch lib?

**Comment by redPanther** [comment]
Can you try without qt5. It's not fully implemented in hyperion stable.

**Comment by NicoHood** [comment]
Does not work for both:
```
  531  cmake -DCMAKE_INSTALL_PREFIX="${pkgdir}/usr" -DPLATFORM=x86 -DCMAKE_BUILD_TYPE=Release -DUSE_SHARED_AVAHI_LIBS=ON ..
  532  cmake -DCMAKE_INSTALL_PREFIX="${pkgdir}/usr" -DPLATFORM=x86 -DCMAKE_BUILD_TYPE=Release -DENABLE_QT5=ON -DUSE_SHARED_AVAHI_LIBS=ON -DUSE_SYSTEM_PROTO_LIBS=ON ..
```

**Comment by NicoHood** [comment]
So the AUR package now works with qt5. It was an error about using the submodule which conflicted with the installed protobuf.

I am also packaging the hyperion-rpi package which has the pwm and ws28xx drivers enabled. Now what would happen if you use this binary on another arm system that has not the pinlayout as the raspberry pi? Would it crash, would it throw an error?

The reason why I ask is simple:
If I could add this as normal ARM build option, we do not need a special package for the raspi etc. and I could just integrate it to the normal hyperion package. On systems who do not have raspi pins the program would just have not those drivers, but adalight for example would still work.

**Comment by redPanther** [comment]
the raspi binary is exclusively for raspberry. For other arm boxes you need to compile from source.

for compilation we will add platform presets for other arm boards too, but this will be available in hyperion v2 (currently the hyperion.ng repo). I just added some support for new amlogic/wetek boxes with 64bit os.

**Comment by NicoHood** [comment]
Yes I saw that. My hope was to have a single binary that works for all arm devices at once. Otherwise I have to create tons of PKGBUILDs.

**Comment by redPanther** [comment]
The fact that we have to deal with 32bit vs. 64bit, arm vs. x86,  lot's of special hardware capabilities and different OS (different linux distros and osx) will make it impossible to have a one fits all binary solution.

But when compiling hyperion.ng can auto determin platform and choose prefered platform settings automaticly. So you have to create only one pkgbuild. This will compile on all platforms (.. we support and tested).

just do not use -DPLATFORM and auto detection will be enabled. But no guarantee that this will work in all cases. This is in a "works for me" state and looking into this is not our current task.
